internal/receiver/discoveryreceiver/statement_evaluator.go

```diff
@@ -83,6 +76,16 @@ func newStatementEvaluator(logger *zap.Logger, id component.ID, config *Config,
 	return se, nil
 }
 
+// exprEnv will unpack logged statement message and field content for expr program use
+func (se *statementEvaluator) exprEnv(pattern string) map[string]any {
```

Comment:
I guess I was thinking that since this method only uses the logger field of statementEvaluator it's not cohesive to this type. Instead, it's used to extend behavior of an evaluator, so perhaps it should live there? (And it could just be a function that accepts a logger.)

Reply:
I may not be following your suggestion since the metric evaluator defines its own env function as is expected w/ derived types: https://github.com/signalfx/splunk-otel-collector/blob/e7064dc00394f18368c6bdcdf86083831be4761c/internal/receiver/discoveryreceiver/metric_evaluator.go#L62